hist/hist/src/TGraph.cxx

```diff
@@ -1545,13 +1545,16 @@ TH1F *TGraph::GetHistogram() const
    if (fNpoints > npt) npt = fNpoints;
    const char *gname = GetName();
    if (!gname[0]) gname = "Graph";
+   // do not add the histogram to gDirectory
+   Bool_t addStatus = TH1::AddDirectoryStatus();
```

Comment:
This is very dubious .. This is a global setting and thus affect the behavior of independent threads (that no longer add the histogram to the current file as the user would expect).